pkg/flink/internal/utils/panic_test.go

```diff
@@ -0,0 +1,18 @@
+package utils
+
+import (
+	"sync"
+	"testing"
+)
+
+func TestPanicRecoveryWrapper(t *testing.T) {
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	go WithPanicRecovery(func() {
+		defer wg.Done()
+		panic("a panic in a goroutine")
+	})()
+
+	wg.Wait()
```

Comment:
Not sure if this is a good test, if you have a bug here the test will wait indefinitely? 

If here what are you testing is the panic recovery, which advantage is this grouting giving to you?

Reply:
Good point, but doesn't the `defer wg.Done()` ensure it's always called? 🤔 
Anyways, I think you're right, it's probably enough to just test the panic recovery behavior without the goroutine. I wanted to prove it also works with goroutines, but that might be unnecessary.

Reply:
ooh right, you had that in a defer. So i guess that's fine, you can leave it as it

Reply:
No, I think you had a good point. I've refactored it a bit and now I can also reuse this wrapper in application.go (where we already had a custom panic recovery )